services/scabbard/libscabbard/src/service/state.rs

```diff
@@ -666,18 +671,25 @@ enum EventQuery {
 /// An iterator that wraps the `TransactionReceiptStore` and returns `StateChangeEvent`s using an
 /// in-memory cache.
 pub struct Events {
+    // transaction_receipt_store is not used if diesel-receipt-store is enabled
+    #[allow(unused_variables)]
     transaction_receipt_store: Arc<RwLock<TransactionReceiptStore>>,
+    #[cfg(feature = "diesel-receipt-store")]
+    receipt_store: Arc<RwLock<dyn ReceiptStore>>,
```

Comment:
These are different enough that I would actually suggest two different implementations of the struct, one for when the feature is enabled and one for `not`.

Reply:
Updated